### PR TITLE
Ensure `KeplerPRF` integrates to the requested flux in the presence of focus parameters

### LINKED ISSUE #219: KeplerPRF is not normalized when `scale_col` or `scale_row` are different from 1
For example, these unit tests fail:
```Python
def test_focus_normalization():
    prf = KeplerPRF(channel=56, shape=[20, 20], column=0, row=0)
    assert np.isclose(prf.evaluate(center_col=10, center_row=10, scale_col=1.5).sum(), 1., rtol=0.01)
    assert np.isclose(prf.evaluate(center_col=10, center_row=10, scale_row=0.5).sum(), 1., rtol=0.01)
```

This explains why @AndyTza is having problems with PRF photometry for AGN!

### PR DESCRIPTION
**Pull request by barentsen**
This PR is an attempt to fix #219.  It makes the integral of `KeplerPRF` significantly more accurate when focus parameters are passed, though the integral of the PRF still only approximates the requested flux to within ~5%. (This may be due to errors introduced by the interpolation step?)

### CODE REVIEW

File: lightkurve/prf/tests/test_prfmodel.py [hunk 1]
```diff
@@ -96,3 +96,13 @@ def test_keplerprf_gradient_against_calculus(param_to_test):
...
+
+
+def test_focus_normalization():
+    """PRF should integrate to one, even if focus parameters are passed."""
+    prf = KeplerPRF(channel=56, shape=[30, 30], column=0, row=0)
+    for scale_col in [0.9, 1.0, 1.1]:
+        for scale_row in [0.9, 1.0, 1.1]:
+                myprf = prf.evaluate(center_col=15, center_row=15, flux=1.,
+                                     scale_col=scale_col, scale_row=scale_row)
+                assert np.isclose(myprf.sum(), 1., rtol=0.05)
```
Comment (barentsen) [inline lightkurve/prf/tests/test_prfmodel.py]: This fails for e.g. `rtol=0.01`

**Comment by mirca** [comment]
Those changes make sense to me. As a result, the model gradients have to be changed as well.

**Comment by barentsen** [comment]
I can't figure out how to fix the gradient =/

**Comment by mirca** [comment]
@barentsen I can do that soonish!

**Comment by mirca** [comment]
@barentsen Travis seems happy, lucky me :D

**Comment by barentsen** [comment]
Yassss!!! :fireworks:   @mirca is the best :+1: :+1:

**Comment by mirca** [comment]
@barentsen actually, could you add unit tests for the gradients for different values of `scale_col`, `scale_row`?

File: lightkurve/prf/tests/test_prfmodel.py [hunk 2]
```diff
@@ -80,7 +80,8 @@ def test_keplerprf_gradient_against_calculus(param_to_test):
     """is the gradient of KeplerPRF consistent with Calculus?
     """
     params = OrderedDict([('center_col', 7), ('center_row', 7), ('flux', 1000.),
-                          ('scale_col', 1.), ('scale_row', 1.), ('rotation_angle', 0)])
+                          ('scale_col', 1.), ('scale_row', 1.), ('rotation_angle', 0),
+                          ('scale_col', 0.8), ('scale_row', 1.2), ('rotation_angle', 1)])
```
[the remark this answers is not in the document]
Comment (barentsen) [reply lightkurve/prf/tests/test_prfmodel.py]: Right, I now see this was a silly thing to do!

**Comment by barentsen** [comment]
@mirca Fixed it!  (Thanks, I finally figured out how `parameterize` works. So simple!)

It's still failing tho =(   I think interpolation errors are at play.

**Comment by mirca** [comment]
The gradients look fine for me, I think the test
```
assert np.max(np.abs(prf_grad[param_order[param_to_test]] - diff_prf) 
                / (1. + np.abs(diff_prf))) < 1e-5
```

is actually a bit too rigorous. We should probably use ``np.mean`` rather than ``np.max`` and increase the tolerance a bit.

**Comment by christinahedges** [comment]
@mirca @barentsen this is quite old but it seems from the comments that you guys came to a conclusion that this should be merged. What should we do with this?

**Pull request closed without merging**